menu/views.py

```diff
@@ -163,18 +163,29 @@ def suggestedcontent(request):
 
     suggested_pages = []
     for topic_id in topics_viewed:
-        for page in (
-            ContentPage.objects.get(id=topic_id)
-            .get_descendants()
-            .filter(numchild=0)
-            .live()
-        ):
-            suggested_pages.append(page)
+        try:
+            pages = (
+                ContentPage.objects.get(id=topic_id)
+                .get_descendants()
+                .filter(numchild=0)
+                .live()
+            )
+        except ContentPage.DoesNotExist:
+            pages = None
+
+        if pages:
+            for page in (
+                ContentPage.objects.get(id=topic_id)
```

Comment:
Looks like this will cause us to make the DB query twice, which will impact performance. Can we use pages here?